src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,617 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integration
+import data.real.ereal
+
+/-!
+
+# Vector valued measures
+
+This file defines vector valued measures, which are σ-additive functions from a set to a add monoid
+`M` such that it maps the empty set and non-measurable sets to zero. In the case
+that `M = ℝ`, we called the vector measure a signed measure and write `signed_measure α`.
+Similarly, when `M = ℂ`, we call the measure a complex measure and write `complex_measure α`.
+
+## Main definitions
+
+* `vector_measure` is a vector valued, σ-additive function that maps the empty
+  and non-measurable set to zero.
+
+## Implementation notes
+
+We require all non-measurable sets to be mapped to zero in order for the extensionality lemma
+to only compare the underlying functions for measurable sets.
+
+We use `has_sum` instead of `tsum` in the definition of vector measures in comparison to `measure`
+since this provides summablity. In the case of `signed_measure`, this is not necessary and one
+can construct a signed measure without proving summable using `signed_measure'.to_signed_measure`.
+
+## Tags
+
+vector measure, signed measure, complex measure
+-/
+
+noncomputable theory
+
+open_locale classical big_operators nnreal ennreal
+
+variables {α β : Type*} [measurable_space α]
+
+/-- A vector measure on a measurable space `α` is a σ-additive over some add monoid `M` is a
+`M`-valued function that maps the empty set to zero. -/
+structure vector_measure (α : Type*) [measurable_space α]
+  (M : Type*) [add_comm_monoid M] [topological_space M] :=
+(measure_of : set α → M)
+(empty : measure_of ∅ = 0)
+(not_measurable ⦃i : set α⦄ : ¬ measurable_set i → measure_of i = 0)
+(m_Union ⦃f : ℕ → set α⦄ :
+  (∀ i, measurable_set (f i)) → pairwise (disjoint on f) →
+  has_sum (λ i, measure_of (f i)) (measure_of (⋃ i, f i)))
+
+-- A `signed_measure` is a `ℝ`-vector measure.
+notation `signed_measure ` α := vector_measure α ℝ
+-- A `complex_measure` is a `ℂ`-vector_measure.
+notation `complex_measure ` α := vector_measure α ℂ
+
+open set measure_theory
+
+namespace vector_measure
+
+variables {M : Type*} [add_comm_monoid M] [topological_space M]
+
+instance : has_coe_to_fun (vector_measure α M) :=
+⟨λ _, set α → M, vector_measure.measure_of⟩
+
+lemma apply (v : vector_measure α M) (i : set α) : v i = v.measure_of i := rfl
+
+@[simp]
+lemma measure_of_empty (v : vector_measure α M) : v ∅ = 0 := v.empty
+
+lemma measure_of_not_measurable_set (v : vector_measure α M)
+  {i : set α} (hi : ¬ measurable_set i) : v i = 0 := v.not_measurable hi
+
+lemma measure_of_disjoint_Union_has_sum (v : vector_measure α M) {f : ℕ → set α}
```

Comment:
the order of words in the lemma name should follow the order in the statement. Since the statement starts with `has_sum`, the lemma name should do the same. Probably `has_sum_of_disjoint_Union`. Same thing several times below.